engine/apps/slack/views.py

```diff
@@ -507,9 +492,36 @@ def _get_organization_from_payload(self, payload, slack_team_identity):
 
     def _open_warning_window_if_needed(self, payload, slack_team_identity, warning_text) -> None:
         if payload.get("trigger_id") is not None:
-            step = ScenarioStep(slack_team_identity)
+            slack_client = SlackClientWithErrorHandling(slack_team_identity.bot_access_token)
+
+            view = {
+                "type": "modal",
+                "callback_id": "warning",
+                "title": {
+                    "type": "plain_text",
+                    "text": ":warning: Warning",
+                },
+                "close": {
+                    "type": "plain_text",
+                    "text": "Ok",
+                    "emoji": True,
+                },
+                "blocks": [
+                    {
+                        "type": "section",
+                        "text": {
+                            "type": "mrkdwn",
+                            "text": warning_text,
+                        },
+                    },
+                ],
+            }
             try:
-                step.open_warning_window(payload, warning_text)
+                slack_client.api_call(
```

Comment:
I wonder why not use `open_warning_window` here?

Reply:
agreed here, this seems to duplicate a lot of the logic in `scenario_step.ScenarioStep.open_warning_window`

Reply:
Reason is same, as in comment above.  I believe, that creating step only to call it's method is a bad practice.

Reply:
On the other hand, this call is inside slack module, so it's kind of fair tradeoff

Reply:
would it make sense to refactor the `open_warning_window` from the `ScenarioStep` to some other Slack related class? imo it's not worth duplicating this logic

Reply:
Decided to return original variant, refactoring ScenarioSteps themselves, like checking which methods are used for slack interactions and so on are topic of different big PR.